persimmon/website/views.py

```diff
@@ -133,3 +133,56 @@ def get_my_accounts(request):
         'balance': account.balance,
         'approval_status': account.approval_status,
     } for account in accounts]
+
+@api_function
+def get_my_address(request):
+    user = current_user(request)
+    return [{
+        'my address': user.address
+    }]
+
+@api_function
+def change_my_address(request, new_address: str):
+    user = current_user(request)
+    user.address = new_address
+    user.save()
+    return [{
+        'my new address': user.address
+    }]
+
+
+#transfer amount from the balance of bankaccount with acountnumb1 id
+@api_function
+def transfer_funds(request, accountnumb1: int, amount: Decimal, accountnumb2: int):
+    user = current_user(request)
+    account1 = BankAccount.objects.filter(owner=user, id=accountnumb1).exclude(approval_status=ApprovalStatus.DECLINED)
```

Comment:
Welcome to security! This introduces a \~race condition\~! If you have multiple queries/transactions that need to be atomic, you need to put them inside a _transaction_.

```
from django.db import transaction

def viewfunc(request):
    # This code executes in autocommit mode (Django's default).
    do_stuff()

    with transaction.atomic():
        # This code executes inside a transaction.
        do_more_stuff()
```

Reply:
I did not think of this at the time, I will add that to my code.